dockstore/dockstore-support
# Aggregate cost metric

**Description**
This PR aggregates the cost metric.

Companion PRs:
- https://github.com/dockstore/dockstore/pull/5577
- https://github.com/dockstore/dockstore-ui2/pull/1805

**Review Instructions**
- Submit a run execution metric that specifies a cost for a workflow. Read this [doc](https://docs.dockstore.org/en/stable/advanced-topics/metrics.html#how-do-i-submit-workflow-metrics) for more info if you're unfamiliar with submitting metrics. You must be a curator or admin to be able to submit metrics.
- Perform a release of the metrics aggregator so that the new JAR is in artifactory. View the [wiki](https://wiki.oicr.on.ca/display/DOC/Dockstore+Releases#DockstoreReleases-RepositoriesthatuseFriendlyCIversionsandaGitHubAction) for how to do this
  - [ ] Note to self: release after this is merged so reviewing is easier for the verifier
 - SSM onto the deployer and wget the metrics aggregator JAR from artifactory
 - Aggregate the metrics. View the [README](https://github.com/dockstore/dockstore-support/tree/develop/metricsaggregator#metrics-aggregator) for instructions.
 - Go to the workflow version that you submitted metrics for in the UI and enable the metrics flag. Verify that the cost metric shows up.

**Issue**
[SEAB-5625](https://ucsc-cgl.atlassian.net/browse/SEAB-5625)

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install` in the project that you have modified (until https://ucsc-cgl.atlassian.net/browse/SEAB-5300 adds multi-module support properly)
- [x] If you are changing dependencies, check with dependabot to ensure you are not introducing new high/critical vulnerabilities


[SEAB-5625]: https://ucsc-cgl.atlassian.net/browse/SEAB-5625?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
